app/src/main/java/org/dharmaseed/android/TalkRepository.java

```diff
@@ -130,43 +135,7 @@ public Cursor getTalksByTeacher(
             boolean isDownloaded
     )
     {
-        String query = "SELECT talks._id, talks.title, talks.teacher_id, teachers.name, talks.rec_date FROM talks";
```

Comment:
After poking around a bit, it seemed possible to re-use the query that gets built in `getTalks()` with only one modification: the addition of an extra WHERE clause to filter on teacher id (the same goes for `getTalksByCenter()` below).  The final query is slightly different (the search terms are more general in `getTalks()`), but I think it's OK because the filter on teacher id will ensure that we don't get extra talks showing up (i.e. #37).  Please let me know if I've missed something though!

Reply:
I think this is a great idea. It may take some tweaking (in the prod code, not the test code), but you should be able to change around `getTalks()` and `getTalksByTeacher()`, etc. without changing any of the unit tests that use them.